corpusreaders/src/main/java/edu/illinois/cs/cogcomp/nlp/corpusreaders/CoNLLGenericReader.java

```diff
@@ -0,0 +1,275 @@
+            spanCollectors.put(column, new SpanCollector());
+        }
+
+        sentences.startSpan(0, ViewNames.SENTENCE);
+
+        for (String line : lines) {
+            line = line.trim();
+
+            int currentTokenId = tokens.size();
+
+            if (line.isEmpty()) {
+                sentences.stopSpan(currentTokenId);
+                sentences.startSpan(currentTokenId, ViewNames.SENTENCE);
+                continue;
+            }
+
+            String[] parts = line.split("\t");
+
+            String token = parts[columnConfig.tokenColumn];
+            if (token.equals("-DOCSTART-") || token.trim().isEmpty()) {
+                continue;
+            }
+            tokens.add(token);
+
+            for (Integer column : columnConfig.tokenLabelColumns) {
+                tokenLabelLists.get(column).add(new Pair<>(currentTokenId, parts[column]));
+            }
+
+            for (Integer column : columnConfig.iobColumns) {
+                String part = parts[column];
+                SpanCollector spanCollector = spanCollectors.get(column);
+                if (part.startsWith("B-") || part.startsWith("U-")) {
+                    // two consecutive entities.
+                    spanCollector.stopSpan(currentTokenId);
+                    spanCollector.startSpan(currentTokenId, part.split("-")[1]);
+                }
+                else if (part.startsWith("I-")) {
+                    // don't do anything....
+                }
+                else if (part.startsWith("O") || part.startsWith("_")){
+                    spanCollector.stopSpan(currentTokenId);
+                }
+            }
+        }
+
+        sentences.stopSpan(tokens.size());
+        for (Integer column : columnConfig.iobColumns) {
+            spanCollectors.get(column).stopSpan(tokens.size());
+        }
+
+
+
+        // we jump through these hoops so we can give the TA an id.
+        String filenameOnly = IOUtils.getFileName(filename);
+        TextAnnotation ta = BasicTextAnnotationBuilder.createTextAnnotationFromTokens(
+                this.corpusName, filenameOnly, Collections.singletonList(tokens.toArray(new String[0])));
+
+        createSpanLabelView(sentences.collect().stream(), ta, ViewNames.SENTENCE, false);
+
+        processViews(ta, tokenLabelLists, spanCollectors);
+
+        return ta;
+    }
+
+    @Override
+    protected void initializeReader() {}
+
+    @Override
+    public boolean hasNext() {
+        return textAnnotations.size() > currentAnnotationId;
+    }
+
+    /**
+     * Return the next annotation object
+     */
+    @Override
+    public TextAnnotation next() {
+        return textAnnotations.get(currentAnnotationId++);
+    }
+
+    /**
+     * TODO: generate a human-readable report of annotations read from the source file
+     * TODO: (plus whatever other relevant statistics the user should know about).
+     */
+    @Override
+    public String generateReport() {
```

Comment:
ideally, please generate a report as you parse the file. Typically, track any non-fatal exceptions and the numbers of documents/tokens processed, and the number of elements in each layer of annotation. This supports the correct reporting of performance of components that model these annotations.